### PR TITLE
Emitter support and reactive dependency updates

### PR DESCRIPTION
**Pull request by cescoffier**
* Rework the emitter support
* Update the Reactive Messaging version and Axle client version
* Update documentation (`ConsumeEvent.blocking`, new axle method)

### CODE REVIEW

File: extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java [hunk 1]
```diff
@@ -124,7 +140,8 @@ public void build(SmallRyeReactiveMessagingTemplate template, BeanContainerBuild
                         .getIdentifier());
             }
         }
-        template.registerMediators(beanClassToBeanId, beanContainer.getValue());
+        template.registerMediators(beanClassToBeanId, beanContainer.getValue(),
+                emitterFields.stream().map(EmitterBuildItem::getName).collect(Collectors.toList()));
```
Comment (mkouba) [inline extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java]: It seems we don't need `EmitterBuildItem.bean` and `EmitterBuildItem.field` at all?

File: extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java [hunk 2]
```diff
@@ -89,6 +93,17 @@ public void validate(ValidationContext validationContext) {
                                 LOGGER.debugf("Found mediator business method %s declared on %s", method, bean);
                             }
                         }
+
+                        for (FieldInfo field : bean.getTarget().get().asClass().fields()) {
+                            if (annotationStore.hasAnnotation(field, NAME_STREAM)) {
+                                if (field.type().name().equals(
+                                        DotName.createSimple(Emitter.class.getName()))) {
```
Comment (mkouba) [inline extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java]: We should create a constant for `Emitter` dot name.

**Review by mkouba — CHANGES_REQUESTED** [review]

File: extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java (hunk 1 above)
Comment (cescoffier) [reply extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java]: you are right, I just need the name.

**Comment by cescoffier** [comment]
@mkouba Code updated.

File: extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java (hunk 2 above)
Comment (cescoffier) [reply extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java]: Done

File: extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java [hunk 3]
```diff
@@ -78,17 +83,26 @@ public void validate(ValidationContext validationContext) {
                 for (BeanInfo bean : validationContext.get(Key.BEANS)) {
                     if (bean.isClassBean()) {
                         // TODO: add support for inherited business methods
-                        for (MethodInfo method : bean.getTarget()
-                                .get()
-                                .asClass()
-                                .methods()) {
+                        ClassInfo ci = bean.getTarget()
+                                .orElseThrow(() -> new IllegalStateException("Target expected")).asClass();
```
Comment (mkouba) [inline extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java]: If `bean.isClassBean()==true` the target is never empty.

File: extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java [hunk 4]
```diff
@@ -78,17 +83,26 @@ public void validate(ValidationContext validationContext) {
...
+                        for (MethodInfo method : ci.methods()) {
                             if (annotationStore.hasAnnotation(method, NAME_INCOMING)
                                     || annotationStore.hasAnnotation(method, NAME_OUTGOING)) {
                                 // TODO: validate method params and return type?
                                 mediatorMethods.produce(new MediatorBuildItem(bean, method));
                                 LOGGER.debugf("Found mediator business method %s declared on %s", method, bean);
                             }
                         }
+
+                        for (FieldInfo field : ci.fields()) {
```
Comment (mkouba) [inline extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java]: Hm, this means that we only support field injection. In fact, we can simply iterate over all injection points:
```java
for (InjectionPointInfo injectionPoint : validationContext.get(Key.INJECTION_POINTS)) {
   // TODO
}
```
I'll try to update this PR in a moment...

**Comment by mkouba** [comment]
@cescoffier Pls review my updates...

**Review by mkouba — APPROVED** [review]

**Pull request merged**